### PR TITLE
fix: component would fail to load when using an instantsearch adapter

### PR DESCRIPTION
**Pull request by djdembeck**
Using an adapter such as [Typesense](https://github.com/typesense/typesense-instantsearch-adapter), which doesn't have the `addAlgoliaAgent` function, could cause this component to not load.

Since this is just for the search user agent, it shouldn't be mission-critical if it fails to change the user agent string:
https://www.algolia.com/doc/api-client/getting-started/customize/javascript/?client=javascript#custom-user-agents

### CODE REVIEW

**Comment by knd775** [comment]
Thanks, I was ready to fork the project over this

File: src/lib/InstantSearch.svelte [hunk 1]
```diff
@@ -36,7 +36,12 @@
 
     search.start();
 
-    searchClient.addAlgoliaAgent("svelte-algolia-instantsearch", "1.0.0");
+    // Try to add a custom user agent, but don't crash on fail since some instantsearch.js adapters don't have this.
+    try {
+        searchClient.addAlgoliaAgent("svelte-algolia-instantsearch", "1.0.0");
+    } catch (e) {
+        // ignore
+    }
```
Comment (aymeric-giraudet) [inline src/lib/InstantSearch.svelte:44]: Hey, thanks for the PR !

```suggestion
    if('addAlgoliaAgent' in searchClient) {
      searchClient.addAlgoliaAgent("svelte-algolia-instantsearch", "1.0.0");
    }
```

I would prefer this, it's shorter and wouldn't dismiss eventual exceptions thrown by an actual `addAlgoliaAgent` function

**Review by aymeric-giraudet — CHANGES_REQUESTED** [review]

File: src/lib/InstantSearch.svelte (hunk 1 above)
Comment (djdembeck) [reply src/lib/InstantSearch.svelte]: I like this much better as well, thanks for the input!

**Comment by djdembeck** [comment]
Hey @aymeric-giraudet just following up if you were able to re-review the changes I accepted. Thanks!

**Comment by aymeric-giraudet** [comment]
Hey @djdembeck, looks good ! Thanks !
I'll put this in the `0.11.0` release :)

**Pull request merged**